### PR TITLE
shfmt: Add package

### PR DESCRIPTION
**Pull request by qwerty12**
[A {bash, mksh, sh} script formatter](https://github.com/mvdan/sh#shfmt)

### CODE REVIEW

File: packages/shfmt/build.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+TERMUX_PKG_HOMEPAGE=https://github.com/mvdan/sh
+TERMUX_PKG_DESCRIPTION="A shell parser, and formatter"
```
Comment (buttaface) [inline packages/shfmt/build.sh:2]: Remove comma.

File: packages/shfmt/build.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+TERMUX_PKG_SHA256=db5d86d9caeb025e41307cb834b21d7a35e171afe5257277e74cb3e44a17ae00
+
+termux_step_make_install() {
+	cd "$TERMUX_PKG_SRCDIR"
+
+	termux_setup_golang
+
+	export GOPATH="$TERMUX_PKG_BUILDDIR"
+	mkdir -p "$GOPATH/src/github.com/mvdan"
+	ln -sf "$TERMUX_PKG_SRCDIR" "$GOPATH/src/github.com/mvdan/sh"
```
Comment (buttaface) [inline packages/shfmt/build.sh:15]: `TERMUX_PKG_BUILD_IN_SRC=true` instead?

Comment (qwerty12) [reply packages/shfmt/build.sh:15]: I considered putting it in, but when looking at the other Go-using build.sh files for the right stuff to copy and paste, they all omit `TERMUX_PKG_BUILD_IN_SRC` and do this `cd`, `GOPATH` etc. stuff instead. After a build, ~/.termux-build/shfmt/src remains identical to the original shfmt source tarball,

Comment (xeffyr) [reply packages/shfmt/build.sh:15]: @buttaface Many Go-based programs require to be built on GOPATH to mimic structure created by `go get`.

**Pull request merged**